app/controllers/incomeSources/add/BusinessAccountingMethodController.scala

```diff
@@ -56,24 +56,28 @@ class BusinessAccountingMethodController @Inject()(val authenticate: Authenticat
                                               (implicit user: MtdItUser[_], backUrl: String, postAction: Call, messages: Messages): Future[Result] = {
     val userActiveBusinesses: List[BusinessDetailsModel] = user.incomeSources.businesses.filterNot(_.isCeased)
 
-    if (userActiveBusinesses.flatMap(_.cashOrAccruals).distinct.size > 1) {
+    if (userActiveBusinesses.flatMap(_.cashOrAccrualsFlag).distinct.size > 1) {
       Logger("application").error(s"${if (isAgent) "[Agent]"}" +
-        s"Error getting business cashOrAccrualsField")
+        s"Error getting business cashOrAccrualsFlag Field")
     }
 
     userActiveBusinesses match {
-      case head :: _ if (head.cashOrAccruals.isDefined) =>
-        val accountingMethod: String = head.cashOrAccruals.get
+      case head :: _ if (head.cashOrAccrualsFlag.isDefined) =>
+        val accountingMethod: String = if (head.cashOrAccrualsFlag.get) {
```

Comment:
if you can move this type transformation to the level of relevant case class => BusinessDetailsModel
boolean => String